aqua/qt5/Portfile

```diff
@@ -1683,15 +1683,28 @@ foreach {module module_info} [array get modules] {
                 # see https://doc.qt.io/qt-5/qtwebengine-platform-notes.html
                 PortGroup           xcodeversion 1.0
 
-                minimum_xcodeversions   {16 8.3.3}
-                if { ${os.major} < 16 } {
+                # see https://trac.macports.org/ticket/61921
+                minimum_xcodeversions   {17 10.0.0}
+                if { ${os.major} < 17 } {
                     known_fail  yes
                     pre-fetch {
-                        ui_error "${subport} requires macOS 10.12 or later"
+                        ui_error "${subport} requires macOS 10.13 or later"
                         return -code error "incompatible OS version"
                     }
                 }
 
+                # build requires nodejs as of 5.15.3
+                # nodejs12 (which chromium 87 uses by default) or later are known to work
+                if { ${os.major} >= 18 } {
+                    # prefer latest LTS nodejs as fallback
+                    depends_build-append bin:node:nodejs16
```

Comment:
this is probably not correct as it will allow any `node` binary in the path (also outside MacPorts) to satisfy the dependency. It should probably be `bin/node:nodejs16` (similar to as is done above for `ninja`, which will allow any `node` in the MacPorts' prefix to satisfy the dependency

Reply:
I should have at least left a comment explaining the `bin:…` rather than `path:…` dependency. While I am aware using non-MacPorts stuff can be undesirable, I may be inclined to leave this as `bin:…` rather than `path:…` unless it is confirmed that only MacPorts’ Node.js will be used for `path:…` dependency. I encountered issues building this port under trace mode (`too many open files` error, which I neglected to report), so trace mode may not be useful for enforcing MacPorts Node.js usage. And because Node.js is not a lightweight dependency (hence why I propose switching it to use known-working nodejs14 on 10.13, rather than unsuccessfully attempt building nodejs16), it seemed desirable to be lenient with how Node.js dependency is provided. See also https://github.com/macports/macports-ports/pull/12595#issuecomment-956960641

Reply:
I don't see an argument in your response to use `bin:node:nodejs16` instead of `bin/node:nodejs16`. Ryan just explained [here](https://github.com/macports/macports-ports/commit/afad77a86ba6be6572cf0aff35db0b13401196f1#commitcomment-61791005) that the former one shouldn't be used.